packages/webview_flutter/lib/platform_interface.dart

```diff
@@ -139,6 +140,12 @@ class WebResourceError {
   ///
   /// This will never be `null` on Android, but can be `null` on iOS.
   final WebResourceErrorType errorType;
+
+  /// Gets the URL for which the resource request was made.
+  ///
+  /// This value is not provided on iOS. Alternatively, you can keep track of
+  /// the last values provided to [WebViewPlatformController.loadUrl].
```

Comment:
I wonder if we can just do that for users on iOS.

Reply:
Hmm.. I considered doing this as well. I don't know if we want to provide a value that the native code doesn't provide just because the other platform does.

Also, Android can provide a URL for any resource, not just the main page. According to [onReceivedError]( https://developer.android.com/reference/android/webkit/WebViewClient#onReceivedError(android.webkit.WebView,%20int,%20java.lang.String,%20java.lang.String)).

 iOS could only provide the last URL used and could return the wrong value if the user navigated to a new webpage without calling `loadUrl`.